pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -272,6 +272,8 @@ func (pl *dynamicResources) EventsToRegister() []framework.ClusterEventWithHint
 		// A resource might depend on node labels for topology filtering.
 		// A new or updated node may make pods schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeLabel}},
+		// A pod might be waiting for a class to get created or modified.
+		{Event: framework.ClusterEvent{Resource: framework.ResourceClass, ActionType: framework.Add | framework.Update}},
```

Comment:
In which scenario rejected Pods should be moved from unsched to backoffQ/activeQ via update events?

Reply:
The node filter in the class might change. I have a hunch that this then leads to an unschedulable pod (no suitable nodes at all) which might be stuck without this update event, already now (i.e. a bug).

Let me write a test for this scenario.

Reply:
It's as I suspected: the [new test](https://github.com/kubernetes/kubernetes/compare/131c81882a07f3c477e8dd580ad6b14ae634665d..a479431782b89c194335e5f61f89147349f221dd) hangs when the `Update` event is not registered.